src/sm.c

```diff
@@ -231,6 +231,16 @@ static SM_RESULT sm_close_begin_internal(SM_HANDLE sm)
                 }
                 else
                 {
+                    if (callback == NULL)
+                    {
+                        // do nothing.
+                    }
+                    else
+                    {
+                        // execute the callback.
+                        callback(callback_context);
+                    }
```

Comment:
shouldn't here be some SRS... bla bla bla??

404 not found.

Reply:
Also I don't see any changes for UT/INT, so that's reason enough to wait.

Reply:
Not sure if we add specs for internal implementations. The reason I added a separate comment instead. I will update the patch with test changes.